Model/Billing/Agreement.php

```diff
@@ -262,6 +276,46 @@ public function setCcBillingAgreement($contractDetail, $storeOneClick, $storeId)
         return $this;
     }
 
+    /**
+     * For sync result to store alternative billing agreement. Currently only sepa is supported. This should be changed
+     * to utilise the factory method for different payment methods
+     *
+     * @param array $additionalData
+     * @param $storeId
+     * @param array $savedPaymentData
+     * @return $this
+     */
+    public function setAlternativePaymentMethodBillingAgreement(array $additionalData, $storeId, array $savedPaymentData): Agreement
+    {
+        $this
+            ->setMethodCode(PaymentMethods::ADYEN_ONE_CLICK)
+            ->setReferenceId($additionalData['recurring.recurringDetailReference']);
+
+        $variant = $additionalData['paymentMethod'];
+
+        $label = __(
+            '%1, %2',
+            $savedPaymentData['ownerName'],
+            $savedPaymentData['iban']
+        );
+
+        $this->setAgreementLabel($label);
+        $recurringType = $this->configHelper->getAlternativePaymentMethodTokenType($storeId);
+
+        $agreementData = [
+            'bank' => [
+                'ownerName' => $savedPaymentData['ownerName'],
+                'iban' => $savedPaymentData['iban'],
+            ],
```

Comment:
Same here. If this is so specific for SEPA I would suggest renaming the function to setSepaBillingAgreement as an alternative for the above.

Reply:
Updated function name [ea60870](https://github.com/Adyen/adyen-magento2/pull/1389/commits/ea608700caec84cf2a23fc8e72a37dd8570f8d7b)